packages/auth/src/mfa/assertions/totp.ts

```diff
@@ -155,10 +155,18 @@ export class TotpMultiFactorAssertionImpl
 
   /** @internal */
   _finalizeSignIn(
-    _auth: AuthInternal,
-    _mfaPendingCredential: string
+    auth: AuthInternal,
+    mfaPendingCredential: string
   ): Promise<FinalizeMfaResponse> {
-    throw new Error('method not implemented');
+    _assert(
+      typeof this.secret !== 'undefined',
```

Comment:
we use secret for enrollment. For sign in, we need to make sure that the enrollmentId and otp are non-empty.